src/webgpu/shader/execution/expression/call/builtin/bitcast.spec.ts

```diff
@@ -71,44 +86,190 @@ const f32InfAndNaNInI32 = f32InfAndNaNInU32.map(u => reinterpretU32AsI32(u));
 const f32ZerosInU32 = [0, kBit.f32.negative.zero];
 const f32ZerosInF32 = f32ZerosInU32.map(u => reinterpretU32AsF32(u));
 const f32ZerosInI32 = f32ZerosInU32.map(u => reinterpretU32AsI32(u));
+const f32ZerosInterval: FPInterval = new FPInterval('f32', -0.0, 0.0);
 
 // f32FiniteRange is a list of finite f32s. fullF32Range() already
 // has +0, we only need to add -0.
 const f32FiniteRange: number[] = [...fullF32Range(), kValue.f32.negative.zero];
 const f32RangeWithInfAndNaN: number[] = [...f32FiniteRange, ...f32InfAndNaNInF32];
 
+// F16 values, finite, Inf/NaN, and zeros. Represented in float and u16.
+const f16FiniteInF16: number[] = [...fullF16Range(), kValue.f16.negative.zero];
+const f16FiniteInU16: number[] = f16FiniteInF16.map(u => reinterpretF16AsU16(u));
+
+const f16InfAndNaNInU16: number[] = [
+  // Cover NaNs evenly in integer space.
+  // The positive NaN with the lowest integer representation is the integer
+  // for infinity, plus one.
+  // The positive NaN with the highest integer representation is u16 0x7fff i.e. 32767.
+  ...linearRange(kBit.f16.infinity.positive + 1, 32767, numNaNs).map(v => Math.ceil(v)),
+  // The negative NaN with the lowest integer representation is the integer
+  // for negative infinity, plus one.
+  // The negative NaN with the highest integer representation is u16 0xffff i.e. 65535
+  ...linearRange(kBit.f16.infinity.negative + 1, 65535, numNaNs).map(v => Math.floor(v)),
+  kBit.f16.infinity.positive,
+  kBit.f16.infinity.negative,
+];
+const f16InfAndNaNInF16 = f16InfAndNaNInU16.map(u => reinterpretU16AsF16(u));
+
+const f16ZerosInU16 = [kBit.f16.negative.zero, 0];
+
+// f16 interval that match +/-0.0.
+const f16ZerosInterval: FPInterval = new FPInterval('f16', -0.0, 0.0);
+
+/**
+ * @returns an u32 whose lower and higher 16bits are the two elements of the
+ * given array of two u16 respectively.
+ */
+function u16x2ToU32(u16x2: number[]): number {
```

Comment:
Both Uint16Array and Uint32Array assume the platform byte order.  (e.g. see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint32Array )

I suspect this code assumes little-endian (which is the most common).
Please use a [DataView](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) for this function and the next one. 

Hey @zoddicus do you know of other endianness sensitivities in the CTS codebase?

Reply:
I see, use the `DataView` to explicitly enforce the little-endian.

All `reinterpret*As*` in `util/conversion.ts` use TypedArray to do the conversion, but most of them use array with one single element so that would be fine. `reinterpretF64AsU32s` and `reinterpretU32sAsF64` assume the little endian, but they are only used for serialization and deserialization for testcases cache, so I thought they would be fine too.

Reply:
I agree with @jzm-intel's analysis.
I would add that all of the AF work is going to assume little endianess for the time being.
Until there is someone with a big endian platform needing bidirectional support, I think it is ok to assume little endianess.